### PR TITLE
Add API testing

### PR DESCRIPTION
**Pull request by mjudeikis**
Test public and private configuration.

@ehashman @asalkeld @jim-minter WDYT?
This makes our testing point to master and tests both private and public configuration using proxy configs.

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml [hunk 1]
```diff
@@ -135,27 +135,51 @@ presubmits:
...
+    rerun_command: /test e2e-create-20191027-private
     spec:
       containers:
       - args:
         - hack/tests/e2e-create.sh
         env:
         - name: API_VERSION
           value: 2019-10-27-preview
+        - name: PRIVATE_API_SERVER
+          value: "true"
```
Comment (asalkeld) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml]: is the proxy server set anywhere? PROXYURL_$REGION

Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml]: it should come from the secrets. I set it in `secrets/secrets`

**Comment by ehashman** [comment]
@mjudeikis my one concern with this CI setup is that we're not running tests against the current public API (e2e-create-20190430)which is probably the most important place to check for regressions -- should we set that to `always_run: true`?

**Comment by mjudeikis** [comment]
> 20190430

Added :)

**Pull request merged**